packages/datetime/src/dateInput.tsx

```diff
@@ -314,7 +316,11 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
                 Utils.safeInvoke(this.props.onChange, this.fromMomentToDate(value));
             }
         } else {
-            this.setState({ isInputFocused: false });
+            if (!valueString.length) {
+                this.setState({ isInputFocused: false, value: moment(null), valueString: null });
```

Comment:
why does `valueString` need to be `null`; can it not be the empty string, or `undefined`?